docs/self-managed/operational-guides/production-guide.md

```diff
@@ -0,0 +1,596 @@
+
+:::note
+The `camunda-credentials` generated secret will not be deleted if the helm chart is uninstalled
+:::
+
+- When upgrading the Camunda Helm Chart, make sure to read the [upgrade guide](/self-managed/operational-guides/update-guide/introduction.md) and corresponding new version elease notes before upgrading and perform the upgrade on a test environment first before attempting in production.
+- Make sure to not store any state or important, long term business data in the local file system of the container. A pod is transient, if the pod is restarted then the data will get wiped. It is better to create a volume and volume mount instead. Here is some example configuration for the core component to create persistent storage:
+
+```yaml
+core:
+  extraVolumes:
+    extraVolumes:
+      - name: persistent-state
+        emptyDir: {}
+    extraVolumeMounts:
+      - name: persistent-state
+        mountPath: /mount
+```
+
+<!-- This seems very specific to the application. I might remove this: -->
+<!-- - Mount Secrets as volumes, not environment variables -->
+
+- It is recommended to set a memory and resource quota for your namespace. Please refer to the [Kubernetes documenation](https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/quota-memory-cpu-namespace/) to do so. Namespace-Level Quotas apply limits to all workloads within a namespace. It ensures aggregate resource consumption by all pods in the namespace do not exceed your desired resource limits.
+
+## Security Guidelines
+
+Here are some points to keep in mind when considering security:
+
+- To limit unnecessary permissions, and reduce risk of token exploitations, it is recommended to disable auto-mounting of the default Service Account. It is possible to do so by adding the following to your `production-values.yaml`:
+
+```yaml
+identity:
+  serviceAccount:
+    enabled: false
+console:
+  serviceAccount:
+    enabled: false
+webModeler:
+  serviceAccount:
+    enabled: false
+connectors:
+  serviceAccount:
+    enabled: false
+core:
+  serviceAccount:
+    enabled: false
+optimize:
+  serviceAccount:
+    enabled: false
+```
+
+You should only enable the auto-mounting of a service account token when the application explicitly needs access to the Kubernetes API server, or you have created a service account with the exact permissions required for the application and bound it to the pod.
+
+- If you have a use case for enabling [Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) then it is recommended to do so.
+<!--Maybe link this to customer: https://github.com/ahmetb/kubernetes-network-policy-recipes-->
+- It is possible to have a pod security standard that is suitable to the security constraints you might have. This is possible through modifying the Pod Security Admission. Please refer to the [Kubernetes documentation](https://kubernetes.io/docs/concepts/security/pod-security-admission/) in order to do so.
```

Comment:
For better readability, in the hyperlinks, let's use the page name instead using the generic name `Kubernetes documentation`.